### ISSUE TITLE
Add a new prop in FormField called labelPrefix

### ISSUE THREAD

**Issue by britt6612** [grommet/grommet#4308]
Currently there is just a storybook example on how you can add an asterisk to a required `formfield` label 
There would be some benefit for users to have this option as a prop instead of making a custom `formfield`
This would make is a little easier for users to use as well as on the design system to have a prop that can be leveraged for other teams when needed.
### Expected Behavior
prop in `formfield` that would allow a asterisk next to a required label 

### Actual Behavior
You have to make a custom `formfield` if you want this behavior 
https://storybook.grommet.io/?path=/story/form--required-label

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by britt6612** [comment]
Adding thread from slack 

currently I have this in the design system above the form
```
const FormFieldRequiredLabel = props => {
  const { required, label, ...rest } = props;
  return (
    <FormField
      label={
        required ? (
          <Box direction="row">
            <Text size="xsmall">{label}</Text>
            <Text size="xsmall">*</Text>
          </Box>
        ) : (
          label
        )
      }
      required={required}
      {...rest}
    />
  );
};
```
which is fine and they can call FormFieldRequiredLabel when they need to show the asterisk however Im just thinking if this would maybe get lost but I go back and forth with adding to grommet or theme since in forms like login and sign up we wouldn't want the asterisk..

so maybe it would be better to just be in guidance on when to use and provide the code for them to copy and paste sorry just rambling my thoughts but I just thought about it when I was making the examples

**Comment by ShimiSun** [comment]
Thanks @britt6612 ! Can you please link to the design-system issue?

**Comment by britt6612** [comment]
https://github.com/hpe-design/design-system/issues/928

**Comment by ShimiSun** [comment]
This is a really good discussion topic. 
The approach of a new prop isn't generic as we would like it to be, and we are thinking that it might need to be added as a prop on the theme  `theme.formField.required` to address the consistent pattern solution (in this case '*'), we'll also need to think through the way that login Form will turn this pattern off.

Few ideas from the discussion:
1. Add `plain` option that will ignore the `formField.required` styles, but consider it will also remove styles of spacing, border, and more of formfield styles, so using `plain` for 'login' might create more work on its FormField styles.
2. Add the type of Form to Form component (similar to kind) so if Form will have a login prop it will handle it accordingly and will ignore the required style and apply styles that are relevant to login Form.

More thought needs to be put in the right solution here, and I'd love to hear what @halocline & @britt6612 has to share on this matter and see if we can extend the ideas further or add new ones to the pool.

@britt6612, in the meanwhile, see if you can simplify the example code on the design-system by simply using `label${required ? ’ *' : ''}`, there might not be a need for the wrapping Box and Text components, let me know if that works.

**Comment by britt6612** [comment]
So initially I was thinking of 2 solutions that we could look at for this issue. 

1. Adding the prop for the label prefix for users to be able to choose if they want to use that as well as what that prefix would be. `*` is just something that we are wanting to use but users could pick which string they want to pass to the label by this prop. I also see your point to not adding in the prop.

2. Since we already have the required prop available to users. I was also thinking of leveraging this prop to be able to take in an object. This would look something like this..
```
required: {prefix: '*'}

```
adding this as an option within the required prop. If a user did not want this behavior they could still do the 
```
required={true}
```
so if a boolean was given to `required` it would do the behavior that is in place now. If a user passed in an object to required it would look for that prefix to add to the label.

**Comment by ShimiSun** [comment]
This issue was taken care of on the `onValidate` PR released on v2.15.0.
